ios/engine/KMEI/KeymanEngine/Classes/Resource Management/LanguagePickAssociator.swift

```diff
@@ -342,7 +344,9 @@ public class LanguagePickAssociator {
   }
 
   public func deselectLanguages(_ languages: Set<String>) {
-    languages.forEach { closureShared.languageSet.remove($0) }
+    closureShared.querySyncQueue.sync {
+      languages.forEach { closureShared.languageSet.remove($0) }
+    }
```

Comment:
Is `languageSetSearch.remove($0)` needed?

Reply:
Absolutely not.  

`languageSetSearch` notes every language that's been queried, not just the ones currently selected.  If a language is in there, noting this facilitates reuse of the originally-triggered query, rather than wastefully running a second or third query for the language.